feature: findtext v1 added

A1exCart3r opened the pull request.

A1exCart3r committed:
feature: findtext v1 added

RuprechtKraus commented on findtext/Program.cs:
```diff
@@ -0,0 +1,47 @@
+    {
+        if (args.Length != 2)
+        {
+            Console.WriteLine("Неверное количество аргументов - Введите аргументы в формате \"findtext.exe \"Евгений Онегин.txt\" \"Я к Вам пишу\"\"");
```
Давай адекватно

A1exCart3r replied:
было исправлено

RuprechtKraus commented on findtext/Program.cs:
```diff
@@ -0,0 +1,47 @@
+        }
+
+        string file = args[0];
+        string lineFromFile;
```
Тоже перенести к месту использования

A1exCart3r replied:
исправил по крайней мере так, как я посчитал нужным

RuprechtKraus commented on findtext/Program.cs:
```diff
@@ -0,0 +1,47 @@
+
+        string file = args[0];
+        string lineFromFile;
+        string searchText = args[1];
```
lineFromFile и searchText лучше местами поменять. Т.к. данные из аргументов друг с другой более связаны

A1exCart3r replied:
переместил

RuprechtKraus commented on findtext/Program.cs:
```diff
@@ -0,0 +1,47 @@
+                    lineNumber++;
+                }
+            }
+            if (!textIsFound) Console.WriteLine("Text Not Found");
```
В большинстве случаев для читабельности лучше не схлопывать так в одну строку, а всегда помещать тело if в скобки

RuprechtKraus replied:
Это и return 0 лучше вынести в конец кода. Не надо раздувать try блоки

A1exCart3r replied:
сделано

RuprechtKraus commented on findtext/Program.cs:
```diff
@@ -0,0 +1,47 @@
+        int lineNumber = 1;
+        bool textIsFound = false;
+
+        if (searchText == "") 
```
Это перенеси на место сразу после проверки аргументов. Если ничего не введено то и переменные выделять не надо

A1exCart3r replied:
сделал

RuprechtKraus commented on findtext/tests.bat:
```diff
@@ -0,0 +1,29 @@
+echo Тест 4 пройден
+
+findtext.exe "У лукоморья дуб зелёный (отрывок из поэмы «Руслан и Людмила»).txt" "run" || goto er
+REM Обычный поиск текста в строке но текст не найден
```
REM на строку выше. А то неконсистентно

A1exCart3r replied:
исправил

RuprechtKraus commented on findtext/tests.bat:
Не хватает тестов:
1. Файл пустой
2. Слово в файле существует, но регистр не совпадает: ищем кот, но в файле написано кОт или КОТ

A1exCart3r replied:
добавил

RuprechtKraus commented:
Смысл задачи в том, чтобы самому написать функцию поиска подстроки в строке. Отделаться использованием стандартного Contains любой может

A1exCart3r committed:
feature: added findtext v2

Задание https://github.com/alexey-malov/oop/tree/master/tasks/01#%D0%B2%D0%B0%D1%80%D0%B8%D0%B0%D0%BD%D1%82-3--findtext--50-%D0%B1%D0%B0%D0%BB%D0%BB%D0%BE%D0%B2-

A1exCart3r committed:
Delete compare/Properties directory

wrong file
